### PR TITLE
capture(ticdc): fix dead look issue

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5150

### What is changed and how it works?
Release the lock immediately after sending the command. Otherwise the initialization and request response will compete for captureMu resulting in a deadlock.

### Check List 

Tests 

 - Integration test

Code changes

None

Side effects

None

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix the issue that requesting `debug/info` API before the completion of capture initialization was stuck
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]
lgtm

**Comment by CharlesCheung96** [comment]
Could you please fix the syntax error in release note?

**Comment by hi-rustin** [comment]
> Could you please fix the syntax error in release note?

How about now?

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -595,11 +596,11 @@ func (c *Capture) WriteDebugInfo(ctx context.Context, w io.Writer) {
 
 	doneM := make(chan error, 1)
 	c.captureMu.Lock()
-	defer c.captureMu.Unlock()
```
Comment (overvenus) [inline cdc/capture/capture.go:598]: Could you add some comments? Also does this problem exist elsewhere?

**Review by overvenus — APPROVED** [review]

**Comment by hi-rustin** [comment]
/hold

Still stuck.

**Pull request merged**